referenceframe/frame.go

```diff
@@ -400,20 +399,88 @@ func (rf *rotationalFrame) Name() string {
 }
 
 func (rf *rotationalFrame) MarshalJSON() ([]byte, error) {
-	m := FrameMapConfig{
-		"type":    "rotational",
-		"name":    rf.name,
-		"rotAxis": rf.rotAxis,
-		"limit":   rf.limits,
-	}
-	return json.Marshal(m)
+	return json.Marshal(rf.toConfig())
+}
+
+func (rf *rotationalFrame) toConfig() FrameMapConfig {
+	config := rf.baseFrame.toConfig()
+	config["type"] = "rotational"
+	config["rotAxis"] = rf.rotAxis
+	return config
 }
 
 func (rf *rotationalFrame) AlmostEquals(otherFrame Frame) bool {
 	other, ok := otherFrame.(*rotationalFrame)
 	return ok && rf.baseFrame.AlmostEquals(other.baseFrame) && spatial.R3VectorAlmostEqual(rf.rotAxis, other.rotAxis, 1e-8)
 }
 
+type linearlyActuatedRotationalFrame struct {
+	*rotationalFrame
+	a, b float64
+}
+
+// NewLinearlyActuatedRotationalFrame creates a frame that represents a robot mechanism where a linear actuator is used to achieve
+// rotational motion about a passive hinge.  This configuration forms a triangle (visualized below) where the linear actuator length is
+// side c, and the offset along the robot links from the rotation is given by a and b, with a being the length along the link previous in
+// the kinematic chain. The transformation given by this frame maps c to theta, which is the angle between a and b
+//
+//                 |
+//                 |
+//                 *
+//                /|
+//            c  / |
+//              /  | b
+//      _______/___|
+//               a
+//
+func NewLinearlyActuatedRotationalFrame(name string, axis spatial.R4AA, limit Limit, a, b float64) (Frame, error) {
```

Comment:
Mmmm, this name is too general a name for what you wrote. I'd wither change the name, or have it take a string to describe the mechanism relation. In this case "telescoping to revolute" or "cherry picker" will work. Then add a case for the law of cosines logic down there. 

Some scissor lifts and reverse rack-pinions can encompassed by this method name. 

Alternatively, why not just create a "newMappedframe" and wrap the math in that? I thought that was a good idea. Juts make the inputs as minimal as possible. I feel like giving a,b might be a lot for the frame system.